go-controller/pkg/ovn/policy.go

```diff
@@ -1404,6 +1404,29 @@ func (oc *DefaultNetworkController) handlePeerPodSelectorDelete(np *networkPolic
 		klog.Infof("Pod %s/%s not scheduled on any node, skipping it", pod.Namespace, pod.Name)
 		return nil
 	}
+
+	if util.PodCompleted(pod) {
+		ips, err := util.GetPodIPsOfNetwork(pod, &util.DefaultNetInfo{})
+		if err != nil {
+			return fmt.Errorf("can't get pod IPs %s/%s: %w", pod.Namespace, pod.Name, err)
+		}
+
+		collidingPod, err := oc.findPodWithIPAddresses(ips)
+		if err != nil {
+			return fmt.Errorf("can't find pods with IP addresses %s: %w", util.JoinIPs(ips, " "), err)
```

Comment:
maybe say something like "lookup for pods with the same IPs failed", because "can't find pods with the same ip" sound like correct behaviour: we don't expect to always find them :)